mastodon/src/main/java/org/joinmastodon/android/ui/text/ClickableLinksDelegate.java

```diff
@@ -90,21 +99,37 @@ public boolean onTouch(MotionEvent event) {
 			}
 		}
 		if(event.getAction()==MotionEvent.ACTION_UP && selectedSpan!=null){
+			longClickHandler.removeCallbacks(copyTextToClipboard);
 			view.playSoundEffect(SoundEffectConstants.CLICK);
 			selectedSpan.onClick(view.getContext());
-			hlPath=null;
-			selectedSpan=null;
-			view.invalidate();
+			resetAndInvalidate();
 			return false;
 		}
 		if(event.getAction()==MotionEvent.ACTION_CANCEL){
-			hlPath=null;
-			selectedSpan=null;
-			view.invalidate();
+			resetAndInvalidate();
 			return false;
 		}
 		return false;
 	}
+
+	Runnable copyTextToClipboard = () -> {
+		//copy link text to clipboard
+		ClipboardManager clipboard = (ClipboardManager) view.getContext().getSystemService(Context.CLIPBOARD_SERVICE);
+		clipboard.setPrimaryClip(ClipData.newPlainText("", selectedSpan.getLink()));
+		//show toast, android from S_V2 on has built-in popup, as documented in
+		//https://developer.android.com/develop/ui/views/touch-and-input/copy-paste#duplicate-notifications
+		if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.S_V2) {
+			Toast.makeText(view.getContext(), "Copied", Toast.LENGTH_SHORT).show();
```

Comment:
This is a translatable string — please don't hardcode it. Either find an existing one in `strings.xml` that fits this case, or add a new one.